prog/ai/inference_endpoint_replica_nexus.rb

```diff
@@ -154,9 +154,46 @@ def ping_gateway
 
     resp = vm.sshable.cmd("sudo curl -s -H \"Content-Type: application/json\" -X POST --data-binary @- --unix-socket /ie/workdir/inference-gateway.clover.sock http://localhost/control", stdin: body.to_json)
     project_usage = JSON.parse(resp)["projects"]
-    # project_usage is a list of the following format:
-    # [{"ubid":"aprojectubid","request_count":0,"prompt_token_count":0,"completion_token_count":0}, ...]
-    # TODO: produce billing records for public endpoints based on project_usage
     Clog.emit("Successfully pinged inference gateway.") { {inference_endpoint: inference_endpoint.ubid, replica: inference_endpoint_replica.ubid, project_usage: project_usage} }
+    update_billing_records(project_usage)
+  end
+
+  def update_billing_records(project_usage)
+    rate = BillingRate.from_resource_properties("InferenceTokens", inference_endpoint.model_name, "global")
+    return if rate.nil? || rate["unit_price"].zero?
+    rate_id = rate["id"]
+    begin_time = Time.now.to_date.to_time
+    end_time = begin_time + 24 * 60 * 60
+
+    project_usage.each do |usage|
```

Comment:
When we receive this data from the gateway, does it reset the token counts? Otherwise, we might repeatedly count tokens that have already been accounted for.